### ISSUE TITLE
Save And load a model 

### ISSUE THREAD

**Issue by zbaida-achraf** [ThilinaRajapakse/simpletransformers#318]
For MultiLabelClassificationModel 
How can i save the model after the trainnig ?
And how i can load it?
I don't know why it takes a lot of time in the training (4H) for a 222K ?

**Comment by ThilinaRajapakse** [comment]
Please read the docs.

The models are saved automatically depending on the parameters you set in args.

You can load the model by providing the path to a saved model directory as `model_name`.

Do you mean for 222K training samples? What is your GPU?

**Comment by Zbaida** [comment]
I know that it save automatically in outputs but i just want to know if i want to use it in other notebook how i can do it (for example torch.save to save a pytorch model) or i need to the full repository ( vocab, model.bin, config.json...)
Yes 222k samples for training. 
I run it in kaggle (NVidia K80 GPUs)

**Comment by Shane-Neeley** [comment]
What is the minimum we need from the outputs to load a saved model for inference only? 

<img width="886" alt="Screen Shot 2020-04-30 at 9 49 18 AM" src="https://user-images.githubusercontent.com/2942715/80737171-fb738b80-8ac7-11ea-9ccd-f48c08e75423.png">

**Comment by ThilinaRajapakse** [comment]
@Zbaida I'm sorry, I seem to have missed your reply.

You don't need to manually `torch.save()`. You can copy the outputs folder to use it with another notebook. You can also copy a specific checkpoint or "best_model" directory. Essentially, any output directory with a `pytorch_model.bin` file is like a saved model.

You can safely delete/ignore the `optimizer.pt` and `scheduler.pt` files if you don't need to continue training from a checkpoint. Those are only needed for resuming training. It's best to keep all the other files though.

**Comment by aaronbriel** [comment]
What is the difference between 'best_model' and the model saved directly under 'outputs'? Is best_model the model that achieved best metrics during one of the epochs (perhaps not the last) while outputs stores the final model?

**Comment by ThilinaRajapakse** [comment]
best_model is the checkpoint with the best evaluation metric. You can prevent checkpoints from being saved by setting `save_eval_checkpoints: False`. However, you can't explicitly prevent the final model from being saved. I don't want to add more saving related flags to control this as I feel like there are already too many such flags.

You could set up your script to delete the final model once the training completes though.

**Comment by aaronbriel** [comment]
Thank you for the quick and detailed response. It is much appreciated!
